google/cloud/bigtable/examples/data_async_snippets.cc

```diff
@@ -124,6 +125,47 @@ void AsyncBulkApply(cbt::Table table, cbt::CompletionQueue cq,
   (std::move(table), std::move(cq), argv[1]);
 }
 
+void AsyncCheckAndMutate(cbt::Table table, cbt::CompletionQueue cq,
+                         std::vector<std::string> argv) {
+  if (argv.size() != 2U) {
+    throw Usage{
+        "async-check-and-mutate: <project-id> <instance-id> <table-id>"};
+  }
+
+  //! [async check and mutate]
+  [](cbt::Table table, cbt::CompletionQueue cq, std::string table_id) {
+    // Check if the latest value of the flip-flop column is "on".
+    auto predicate = google::cloud::bigtable::Filter::Chain(
+        google::cloud::bigtable::Filter::ColumnRangeClosed("fam", "flip-flop",
+                                                           "flip-flop"),
+        google::cloud::bigtable::Filter::Latest(1),
+        google::cloud::bigtable::Filter::ValueRegex("on"));
+    google::cloud::future<google::cloud::StatusOr<
+        google::bigtable::v2::CheckAndMutateRowResponse>>
+        future = table.AsyncCheckAndMutateRow(
+            MAGIC_ROW_KEY, std::move(predicate),
+            {google::cloud::bigtable::SetCell("fam", "flip-flop", "off"),
+             google::cloud::bigtable::SetCell("fam", "flop-flip", "on")},
+            {google::cloud::bigtable::SetCell("fam", "flip-flop", "on"),
+             google::cloud::bigtable::SetCell("fam", "flop-flip", "off")},
+            cq);
+
+    auto final =
+        future.then([](google::cloud::future<google::cloud::StatusOr<
+                           google::bigtable::v2::CheckAndMutateRowResponse>>
+                           f) {
+          auto row = f.get();
+          if (!row) {
+            throw std::runtime_error(row.status().message());
+          }
+
+          return google::cloud::Status();
+        });
+    final.get();
+  }
+  //! [async check and mutate]
+  (std::move(table), std::move(cq), argv[1]);
```

Comment:
What do you think about using the [`ConsumeArg`](https://github.com/googleapis/google-cloud-cpp/blob/e8e461a3416d339cde9948fb2f785ce7dc22e885/google/cloud/bigtable/examples/data_snippets.cc#L39) function present in the other sample files for consistency?

Reply:
I agree, but I do not want to start changing things around unless approved (@coryan).

Reply:
Another case where we should probably make a pass and remove all the `ConsumeArg` things.